src/alcaramba/Drawable/Town/Town.gd

```diff
@@ -72,37 +76,95 @@ func _input(event):
 			if event.button_index == 1 && _placement_mode == 2:
 				# if cell not empty and cell is not starting tile and removed does not break connection
 				var cell_to_remove = get_cell(x, y)
-				if cell_to_remove != TileMap.INVALID_CELL && cell_to_remove != _starting_tile_id && is_tile_removable(x, y):
-					var removed_tile_id = remove_tile(x, y)
+				if cell_to_remove != TileMap.INVALID_CELL && cell_to_remove != _starting_tile_id && _is_tile_removable(x, y):
+					var removed_tile_id = remove_tile(x, y) # remove tile and return its id
 					# TODO #19: send tile to spare tiles
 					draw_placed_tiles()
+				else: 
+					print_debug("cell not removable")
+					draw_placed_tiles()
+				
 
-
+# adds card to TileMap and town stack
+# @param tile: - TileCard received from market/spare tiles
+# @param x: - TileMap local x coordinate
+# @param y: - TileMap local y coordinate
 func place_tile(x: int, y: int, tile: TileCard):
-	# if is_placement_valid(x,y,tile.get_id()):
 	set_cell(x, y, tile.get_id())
 	_town_tiles.add_card(tile)
 	emit_signal("tile_placed", tile, Vector2(x, y))
 
+# remove tile from TileMap and town stack and returns its id
+# @param x: - TileMap local x coordinate
+# @param y: - TileMap local y coordinate
 func remove_tile(x: int, y: int) -> int:
 	var tile_id = get_cell(x, y)
-	_stack_tiles.remove_card_by_id(tile_id)
+	_town_tiles.remove_card_by_id(tile_id)
 	set_cell(x, y, TileMap.INVALID_CELL)
 	print_debug("Removed tile %d at (%d|%d)" % [tile_id, x, y])
 	return tile_id
 
 # check if removing this tile would break continouity of town
-func is_tile_removable(x: int, y: int):
-	# TODO #15: logic for continuity
-	return true
+# @param x: - TileMap local x coordinate
+# @param y: - TileMap local y coordinate
+func _is_tile_removable(x: int, y: int) -> bool:
+	
+	var removal_valid = true
+	var test_id = get_cell(x, y)
+	
+	# if tile is not populated or starting tile, removal not valid
+	if test_id == TileMap.INVALID_CELL || test_id == _starting_tile_id: return false
+	
+	# check if removal would leave hole, happens if all four neighbours are present
+	if _count_neighbours(x, y) == 4: 
+		return false
+	
+	# remove test tile and neighbour then check if neighbour placement is still valid
+	set_cell(x, y, TileMap.INVALID_CELL)
+	
+	# if any tile has infinite distance, the connection to start was broken
+	_update_distances()
+	if !_distances.get_used_cells_by_id(_max_int).empty():
+		removal_valid = false
+	
+	# loop over each neighbour,
+	for _x in range(-1, 2):
```

Comment:
There are multiple instances of exactly the same loop code over neighbours. We should unify this, as this can lead to very weird bugs if it needs to be changed and is only changed in one place.

Reply:
Thats why I implemented get_neighbours function. Planned to refactor this before pull request ;)